pkg/indicator/ghfilter_test.go

```diff
@@ -0,0 +1,6151 @@
+    "low": 1371.96,
+    "close": 1376.76
+  },
+  {
+    "open": 1376.77,
+    "high": 1377.59,
+    "low": 1370.81,
+    "close": 1370.95
+  },
+  {
+    "open": 1370.95,
+    "high": 1374.62,
+    "low": 1363.87,
+    "close": 1367.88
+  },
+  {
+    "open": 1367.88,
+    "high": 1372,
+    "low": 1365.03,
+    "close": 1368.68
+  },
+  {
+    "open": 1368.67,
+    "high": 1373.02,
+    "low": 1367,
+    "close": 1370.9
+  }
+]`)
+
+func Test_GHFilter(t *testing.T) {
+	type args struct {
+		allKLines []types.KLine
+		window    int
+	}
+	var klines []types.KLine
+	if err := json.Unmarshal(testGHFilterDataEthusdt5m, &klines); err != nil {
+		panic(err)
+	}
+	tests := []struct {
+		name string
+		args args
+		want float64
+	}{
+		{
+			name: "ETHUSDT G-H Filter 7",
+			args: args{
+				allKLines: klines,
+				window:    7,
+			},
+			want: 1373.71,
+		},
+		{
+			name: "ETHUSDT G-H Filter 25",
+			args: args{
+				allKLines: klines,
+				window:    25,
+			},
+			want: 1376.21,
+		},
+		{
+			name: "ETHUSDT G-H Filter 99",
+			args: args{
+				allKLines: klines,
+				window:    99,
+			},
+			want: 1378.96,
+		},
+	}
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			filter := &GHFilter{IntervalWindow: types.IntervalWindow{Window: tt.args.window}}
+			for _, k := range klines {
+				filter.PushK(k)
+			}
+			got := filter.Last()
+			got = math.Trunc(got*100.0) / 100.0
+			if got != tt.want {
+				t.Errorf("GHFilter.Last() = %v, want %v", got, tt.want)
+			}
+		})
+	}
+}
+
+func Test_GHFilterEstimationAccurate(t *testing.T) {
+	type args struct {
+		allKLines []types.KLine
+		priceF    KLineValueMapper
+		window    int
+	}
+	var klines []types.KLine
+	if err := json.Unmarshal(testGHFilterDataEthusdt5m, &klines); err != nil {
+		panic(err)
+	}
+	tests := []struct {
+		name string
+		args args
+		want float64
+	}{
+		{
+			name: "ETHUSDT G-H Filter square error 7",
+			args: args{
+				allKLines: klines,
+				window:    7,
+			},
+		},
+		{
+			name: "ETHUSDT G-H Filter square error 25",
+			args: args{
+				allKLines: klines,
+				window:    25,
+			},
+		},
+		{
+			name: "ETHUSDT G-H Filter square error 99",
+			args: args{
+				allKLines: klines,
+				window:    99,
+			},
+		},
+	}
+	klineSquareError := func(base float64, k types.KLine) float64 {
+		openDiff := math.Abs(k.Open.Float64() - base)
+		highDiff := math.Abs(k.High.Float64() - base)
+		lowDiff := math.Abs(k.Low.Float64() - base)
+		closeDiff := math.Abs(k.Close.Float64() - base)
+		return openDiff*openDiff + highDiff*highDiff + lowDiff*lowDiff + closeDiff*closeDiff
+	}
+	closeSquareError := func(base float64, k types.KLine) float64 {
+		closeDiff := math.Abs(k.Close.Float64() - base)
+		return closeDiff * closeDiff
+	}
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			filter := &GHFilter{IntervalWindow: types.IntervalWindow{Window: tt.args.window}}
+			ewma := &EWMA{IntervalWindow: types.IntervalWindow{Window: tt.args.window}}
```

Comment:
EWMA is widely used to calculate average price with less latency compare to SMA. Do we expect the comparison of ZLEMA?